django/db/backends/sqlite3/introspection.py

```diff
@@ -242,21 +243,62 @@ def get_constraints(self, cursor, table_name):
             # table_name is a view.
             pass
         else:
-            fields_with_check_constraints = [
-                schema_row.strip().split(' ')[0][1:-1]
-                for schema_row in table_schema.split(',')
-                if schema_row.find('CHECK') >= 0
-            ]
-            for field_name in fields_with_check_constraints:
-                # An arbitrary made up name.
-                constraints['__check__%s' % field_name] = {
-                    'columns': [field_name],
-                    'primary_key': False,
-                    'unique': False,
-                    'foreign_key': False,
-                    'check': True,
-                    'index': False,
-                }
+            try:
+                import sqlparse
+            except ImportError:
+                raise ImproperlyConfigured(
+                    "The sqlparse package is required to introspect constraints on sqlite.",
+                )
+
+            Token = sqlparse.tokens
+
+            _constraint = False
+            _check = False
+            _unique = False
+            _name = None
+            _columns = []
+
+            def update_constraints():
+                if _constraint or _check:
+                    constraints[_name] = {
+                        'columns': _columns,
+                        'primary_key': False,
+                        'unique': _unique,
+                        'foreign_key': False,
+                        'check': _check,
+                        'index': False,
+                    }
+
+            statement = sqlparse.parse(table_schema)[0]
+            for token in statement.flatten():
```

Comment:
Thanks for the spot. I had help with the parsing logic from @MarkusH.

Reply:
Simon, did you fix this issue in your branch? Feel free to push an update here.

Reply:
Cool I'll do that later tonight.

Reply:
Thanks. There might be a bug because when I rebase https://github.com/django/django/pull/10337, `self.assertIn('unique_name', constraints)` fails on SQLite.

Reply:
I know what's causing it. Let me rebase #10337 on top of this branch. That's where the `token.match(Token.Keyword, 'UNIQUE')` should live anyway.

Reply:
I did the rebase already, shall I push it to Ian's branch?

Reply:
yeah please do, that'll save me doing the rebase.